[doc] [runtime env] Add hint for `pip install` command line options (#39183)
ray-project/ray: doc/source/ray-core/handling-dependencies.rst

You may also specify your ``pip`` dependencies either via a Python list or a local ``requirements.txt`` file.

> Specifying a ``requirements.txt`` file is useful when your ``pip install`` command would require options such as ``--extra-index-url`` or ``--find-links``; see `<https://pip.pypa.io/en/stable/reference/requirements-file-format/#>`_ for details.

Reviewer: ```suggestion
Consider specifying a ``requirements.txt`` file when your ``pip install`` command requires options such as ``--extra-index-url`` or ``--find-links``; see `<https://pip.pypa.io/en/stable/reference/requirements-file-format/#>`_ for details.
```

Reviewer: If it's not obvious why it might be useful, maybe you can add a reason, briefly.
